src/flyte/remote/_run.py

```diff
@@ -57,9 +65,10 @@ async def listall(
             direction=(list_pb2.Sort.ASCENDING if sort_by[1] == "asc" else list_pb2.Sort.DESCENDING),
         )
         cfg = get_common_config()
+        i = 0
         while True:
             req = list_pb2.ListRequest(
-                limit=100,
+                limit=limit,
```

Comment:
I think we can use `min(limit, 100)` here. If we try to fetch too many runs (1000+) in a single gRPC call, it might exceed the gRPC size limit.